src/base/NekRSProblem.C

```diff
@@ -694,20 +851,23 @@ NekRSProblem::syncSolutions(ExternalProblem::Direction direction)
       if (_volume && _has_heat_source)
         sendVolumeHeatSourceToNek();
 
-      // copy the boundary heat flux and/or volume heat source in the scratch space to device
-      nekrs::copyScratchToDevice(_minimum_scratch_size_for_coupling);
 
-      if (_moving_mesh)
+      if (nekrs::hasUserMeshSolver() && _volume)
       {
-        if (_volume)
-        {
-          sendVolumeDeformationToNek();
-          nekrs::copyDeformationToDevice();
-        }
-
-        // no boundary-based mesh movement available in nekRS yet
+        sendVolumeDeformationToNek();
+      }
+      else if (nekrs::hasElasticitySolver())
+      {
+        sendBoundaryDeformationToNek();
       }
 
+      // copy the boundary heat flux and/or volume heat source in the scratch space to device
```

Comment:
Should this comment technically be updated to something like "copy the boundary heat flux, volume heat source, and/or volume mesh displacements in the scratch space to device"?

Reply:
Done, thanks!